teos-common/src/lightning/ser_utils.rs

```diff
@@ -0,0 +1,200 @@
+//! A helper module containing some lightning messages serialization stuff.
+//! Most of this file is taken/inspired from [here](https://github.com/lightningdevkit/rust-lightning/blob/3676a056c85f54347e7e079e913317a79e26a2ae/lightning/src/util/ser.rs).
+
+/* This file is licensed under either of
+ *  Apache License, Version 2.0, (LICENSE-APACHE or http://www.apache.org/licenses/LICENSE-2.0) or
+ *  MIT license (LICENSE-MIT or http://opensource.org/licenses/MIT)
+ * at your option.
+*/
+
+use crate::appointment::{Locator, LOCATOR_LEN};
+
+use lightning::io::{copy, sink, Error, Read};
+use lightning::ln::msgs::DecodeError;
+use lightning::util::ser::{MaybeReadable, Readable, Writeable, Writer};
+
+/// A trait that associates a u16 [`Type::TYPE`] constant with a lightning message.
+pub trait Type {
+    /// The type identifying the message payload.
+    const TYPE: u16;
+}
```

Comment:
Does this mas to be a trait? How does LDK implement this for the standard messages?

Reply:
Might work as something else, but I just followed LDK on this one. They have that trait [`Encode`](https://github.com/lightningdevkit/rust-lightning/blob/main/lightning/src/ln/wire.rs#L267) which does the some thing. Encode is a little bit of a weird name, but I think they named it that way because they have another trait already called `Type`.